crates/printer/src/standard.rs

```diff
@@ -1540,7 +1542,12 @@ impl<'a, M: Matcher, W: WriteColor> StandardImpl<'a, M, W> {
         if self.in_color_match.get() {
             return Ok(());
         }
-        self.wtr().borrow_mut().set_color(self.config().colors.matched())?;
+        let color_spec = if self.highlight_on() {
+            self.match_spec_with_highlight()
+        } else {
+            self.config().colors.matched().clone()
+        };
```

Comment:
This is FUBAR. In `match_spec_with_highlight`, you're doing shenanigans based on whether specific stylings are set. That's no good and it assumes too much about how the user's _intention_ when using `highlight`. If the user wants the background color to be consistent, for example, then they'll need to set it for both the `matched` and `highlight` types.